pkg/virt-operator/resource/generate/components/prometheus.go

```diff
@@ -417,16 +417,12 @@ func NewPrometheusRuleSpec(ns string, workloadUpdatesEnabled bool) *v1.Prometheu
 							severityAlertLabelKey: "warning",
 						},
 					},
-					{
-						Record: "kubevirt_num_virt_handlers_by_node_running_virt_launcher",
-						Expr:   intstr.FromString("count by(node)(node_namespace_pod:kube_pod_info:{pod=~'virt-launcher-.*'} ) * on (node) group_left(pod) (1*(kube_pod_container_status_ready{pod=~'virt-handler-.*'} + on (pod) group_left(node) (0 * node_namespace_pod:kube_pod_info:{pod=~'virt-handler-.*'} ))) or on (node) (0 * node_namespace_pod:kube_pod_info:{pod=~'virt-launcher-.*'} )"),
-					},
 					{
 						Alert: "OrphanedVirtualMachineInstances",
-						Expr:  intstr.FromString("(kubevirt_num_virt_handlers_by_node_running_virt_launcher) == 0"),
-						For:   "60m",
+						Expr:  intstr.FromString("((count by (node) ( kube_pod_info{pod=~'virt-launcher.*'} )*0+ on(node) group_left() (kube_pod_status_ready{condition='true',pod=~'virt-handler.*'} * on(pod) group_left(node) kube_pod_info{pod=~'virt-handler.*'} ))  *on(node) group_left(pod) (count by (node)((kube_pod_info{pod=~'virt-handler.*'} ) ) ) or (count by (node) ( kube_pod_info{pod=~'virt-launcher.*'} )*0) ) == 0"),
```

Comment:
Hey Shirly,
First of all i see that this expression appear `* on(pod) group_left(node) kube_pod_info{pod=~'virt-handler.*'} ))` twice:
`...* on(pod) group_left(node) kube_pod_info{pod=~'virt-handler.*'} ))  *on(node) group_left(pod) (count by (node)((kube_pod_info{pod=~'virt-handler.*'}....`

Also i can't see the logic in here .
Why would this expression  return zero if there isn't virt-handler pod on one of the  cluster's nodes
Can you explain please?

Reply:
@Barakmor1 the 2 difference from the query you created are:
1. Checking for ready virt-handler pods instead of unready virt-handler pods
2. Adding the "or" condition to return 0 in case there is no ready pod or no pod at all.

Reply:
Nice idea ,
although i think that we can simplify the query to:

` (count by (node) (kube_pod_status_ready{condition='true',pod=~'virt-handler.*'} * on(pod) group_left(node) kube_pod_info{pod=~'virt-handler.*'}) ) or (count by (node)( kube_pod_info{pod=~'virt-launcher.*'})*0) `

Reply:
@Barakmor1 This wiil return a list off all nodes that includes a virt-hanlder even if there are no running VMs on the node.